homework_7/hw_fun.py

```diff
@@ -0,0 +1,80 @@
+import sys
+
+
+# Sequential map
+
+def sequential_map(*args) -> list:
+    args = [*args]
+    cont = args.pop()
+    for fun in args:
+        cont = list(map(fun, cont))
+    return cont
+
+
+# Consensus filter
+
+def consensus_filter(*args) -> list:
+    args = [*args]
+    cont = args.pop()
+    for fun in args:
+        cont = list(filter(fun, cont))
+    return cont
+
+
+# Conditional reduce
+
+def conditional_reduce(fun_1, fun_2, cont) -> int:
+    filt_cont = iter(filter(fun_1, cont))
```

Comment:
Результат filter это уже итератор и он умеет работать с `next`, так что `iter` тут не нужен
```suggestion
    filt_cont = filter(fun_1, cont)
```